tests/func/test_api.py

```diff
@@ -56,7 +56,9 @@ def test_get_url_requires_dvc(tmp_dir, scm):
         api.get_url("foo", repo=f"file://{tmp_dir}")
 
 
-@pytest.mark.parametrize("remote_url", all_remote_params, indirect=True)
+@pytest.mark.parametrize(
+    "remote_url", all_remote_params + [GDrive], indirect=True
```

Comment:
Unless I'm missing something, this change should make all the tests in this test_api.py work with Gdrive.

Reply:
nope because of https://github.com/iterative/dvc/blob/edfa1741db506ba12ee0244452349c1db059a026/tests/remotes.py#L144 requiring an argument (`self`).

Reply:
the only other classes that have that are `tests.remotes.HTTP` -> `TestHTTPRemote` (which doesn't actually look used?) and `tests.remotes.SSHMocked`

Reply:
@casperdcl That's some primitive caching that was added. All other get_url's just generate a new one each time, we could try to do the same.